lib/json/json.c

```diff
@@ -714,7 +673,38 @@ ssize_t json_escape(char *str, size_t *len, size_t buf_size)
 		return -ENOMEM;
 	}
 
-	return json_escape_buf(str, len, escaped_len);
+	/*
+	 * By walking backwards in the buffer from the end positions
+	 * of both the original and escaped strings, we avoid using
+	 * extra space. Characters in the original string are
+	 * overwritten only after they have already been escaped.
+	 */
+	str[escaped_len] = '\0';
+	next = &str[*len - 1];
+	dest = &str[escaped_len - 1];
+	while (1) {
```

Comment:
I tried and failed to think of a way to do it without while(1) / break. The edge cases are a little tricky.

The way you suggest won't work if `*len == 1` and the character at `str[0]` needs escaping, for example.